### ISSUE TITLE
Transforms support for ES|QL queries or an alternative

### ISSUE THREAD

**Issue by grabowskit** [elastic/elasticsearch#111599]
### Description

SLOs and Entities rely on Transforms. Transforms do not currently have support for ES|QL. More and more customers are asking for support for support for ES|QL.

SLI definition to include ES|QL has been discussed in a [separate ticket](https://github.com/elastic/kibana/issues/182995)

There have been discussions about using "materialized views", but it is not clear how and when that would be available. We would like to get to a plan on whether to pursue ES|QL in Transforms or in future Materialized views.

cc: @simianhacker @vinaychandrasekhar

**Comment by jasonrhodes** [comment]
For what it's worth, I had a good conversation with @dimkots and the ML team about this a couple months back, which led to the creation of the ticket you linked above as well as this one from the ML + Elasticsearch side: https://github.com/elastic/elasticsearch/issues/108640

I do think that we really need to clarify what we want on the SLO side in order to help determine what kind of "ES|QL support" we want and need from Elasticsearch + ES transforms and/or materialized views.

**Comment by jasonrhodes** [comment]
Yep that's the one I linked to in the comment above, jinx Tom :D
